### PR TITLE
support glob keyword in configfile

### PR DESCRIPTION
**Pull request by balrok**
This implements https://github.com/c4urself/bump2version/issues/64

### CODE REVIEW

File: bumpversion/cli.py [hunk 1]
```diff
@@ -352,9 +353,11 @@ def _load_configuration(config_file, explicit_config, defaults):
 
             if "replace" not in section_config:
                 section_config["replace"] = defaults.get("replace", "{new_version}")
-
-            files.append(ConfiguredFile(filename, VersionConfig(**section_config)))
-
+            if section_type.get("file") == "glob":
```
Comment (ekohl) [inline bumpversion/cli.py:356]: Please keep an empty line in between
```suggestion

            if section_type.get("file") == "glob":
```

File: bumpversion/cli.py [hunk 2]
```diff
@@ -352,9 +353,11 @@ def _load_configuration(config_file, explicit_config, defaults):
...
+                for filename_glob in glob.glob(filename):
+                    files.append(ConfiguredFile(filename_glob, VersionConfig(**section_config)))
```
Comment (ekohl) [inline bumpversion/cli.py:358]: Perhaps `VersionConfig(**section_config)` should be stored in a variable before the if section and used in both branches.

**Comment by balrok** [comment]
thanks for your comments in both PR :-)
having the VersionConfig created outside of the loop will also reuse the object between the globbed-files

**Comment by ekohl** [comment]
>  having the VersionConfig created outside of the loop will also reuse the object between the globbed-files

Yes. I think that's good because it only sets variables in the constructor, not at runtime. This saves some resources.

**Comment by balrok** [comment]
just one small fix, I was not aware that python-glob does not interpret "**" by default as recursive.. one has to set the recursive-flag in this function before.. there is now also a test for this

**Review by ekohl — APPROVED** [review]
Going to merge this since there was no objection from another maintainer and it looks good to me.

**Comment by ekohl** [comment]
Thanks @balrok!

**Comment by lowell80** [comment]
Awesome feature, thanks @balrok!   You just saved me a bunch of time!

**Pull request merged**